src/contracts/core/DelegationManager.sol

```diff
@@ -528,131 +633,224 @@ contract DelegationManager is
     }
 
     /**
-     * @dev commented-out param (middlewareTimesIndex) is the index in the operator that the staker who triggered the withdrawal was delegated to's middleware times array
-     * This param is intended to be passed on to the Slasher contract, but is unused in the M2 release of these contracts, and is thus commented-out.
+     * @dev This function completes a queued withdrawal for a staker.
+     * This will apply any slashing that has occurred since the scaledShares were initially set in the Withdrawal struct up
+     * until the totalMagnitude at completion time.
+     * If receiveAsTokens is true, then these shares will be withdrawn as tokens.
+     * If receiveAsTokens is false, then they will be redeposited according to the current operator the staker is delegated to,
+     * and added back to the operator's scaledShares.
      */
     function _completeQueuedWithdrawal(
         Withdrawal calldata withdrawal,
         IERC20[] calldata tokens,
-        uint256, /*middlewareTimesIndex*/
         bool receiveAsTokens
     ) internal {
         bytes32 withdrawalRoot = calculateWithdrawalRoot(withdrawal);
 
         require(
             pendingWithdrawals[withdrawalRoot], "DelegationManager._completeQueuedWithdrawal: action is not in queue"
         );
-
         require(
-            withdrawal.startBlock + minWithdrawalDelayBlocks <= block.number,
-            "DelegationManager._completeQueuedWithdrawal: minWithdrawalDelayBlocks period has not yet passed"
+            withdrawal.startTimestamp + minWithdrawalDelay <= block.number,
+            "DelegationManager._completeQueuedWithdrawal: minWithdrawalDelay period has not yet passed"
         );
-
         require(
             msg.sender == withdrawal.withdrawer,
             "DelegationManager._completeQueuedWithdrawal: only withdrawer can complete action"
         );
+        require(
+            tokens.length == withdrawal.strategies.length,
+            "DelegationManager._completeQueuedWithdrawal: input length mismatch"
+        );
+
+        // read delegated operator's totalMagnitudes at time of withdrawal to scale shares again if any slashing has occurred
+        // during withdrawal delay period
+        uint64[] memory totalMagnitudes = allocationManager.getTotalMagnitudesAtTimestamp({
+            operator: withdrawal.delegatedTo,
+            strategies: withdrawal.strategies,
+            timestamp: withdrawal.startTimestamp + SlashingConstants.DEALLOCATION_DELAY
```

Comment:
don't use `minWithdrawalDelay` and `SlashingConstants.DEALLOCATION_DELAY`

in fact, having the delay in the library is problematic for different envs

Reply:
So if we have this value set as immutables, we want to make sure that DEALLOCATION_DELAY is the equal to the DelegationManagers WITHDRAWAL_DELAY.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/679/commits/a70ed62f13fa8be58c456c8d518ec71e7a8482ab
Updated here